llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -9076,12 +9198,134 @@ ScalarEvolution::ExitLimit ScalarEvolution::computeExitLimitFromICmp(
 
   const SCEV *LHS = getSCEV(ExitCond->getOperand(0));
   const SCEV *RHS = getSCEV(ExitCond->getOperand(1));
+  if (!AssumeLoopExists) {
+    ExitLimit EL = computeExitLimitFromICmp(L, Pred, LHS, RHS, ControlsOnlyExit,
```

Comment:
This code should probably be integrated in computeExitLimitFromICmp?

Reply:
Things to integrate into `computeExitLimitFromICmp`

at the start of the function, if `AssumeLoopExits` do the same prop_phi stuff.

and also in switch statement add support for LE / GE like below